tv-app/effecticeTV/app/src/main/java/band/effective/office/tv/domain/models/Employee/EmployeeInfoEntity.kt

```diff
@@ -1,8 +1,44 @@
 package band.effective.office.tv.domain.models.Employee
 
+import band.effective.office.tv.utils.DateUtlils
+
 class EmployeeInfoEntity(
     val firstName: String,
     val startDate: String,
     val nextBirthdayDate: String,
     val photoUrl: String,
-)
\ No newline at end of file
+)
+
+fun processEmployeeInfo(data: List<EmployeeInfoEntity>): MutableList<EmployeeInfo> {
```

Comment:
maybe that's better:
fun processEmployeeInfo(data: List<EmployeeInfoEntity>): List<EmployeeInfo> =
    data.map {
        when {
            DateUtlils.isCelebrationToday(it.nextBirthdayDate) ->
                Birthday(
                    it.firstName,
                    it.photoUrl,
                )
            DateUtlils.isCelebrationToday(it.startDate) ->
                Anniversary(
                    it.firstName,
                    it.photoUrl,
                    DateUtlils.getYearsWithTheCompany(it.startDate)
                )
            DateUtlils.isNewEmployeeToday(it.startDate) -> NewEmployee(
                it.firstName,
                it.photoUrl,
            )
            else -> // return empty EmployeeInfo
        }
    }

Reply:
I think it`s not. Because after that, i use this data in ui layer, empty objects can possibly cause more exceptions. Beyond that, I find it generally not logical. If the description of the employee does not fit the description of any type of event, why create an event model at all?

Reply:
This remark is related to the fact that you used the map method strangely. To have only correct employee, you can use filter